### ISSUE TITLE
Gallery examples use Matplotlib in a hybrid way (explicit vs. implicit interfaces).

### ISSUE THREAD

**Issue by mkcor** [scikit-image/scikit-image#7330]
### Description:

I would suggest using the "object-oriented (OO) style" (as opposed to pyplot-style) across the gallery.

This issue replicates https://github.com/datacarpentry/image-processing/issues/319, which I opened for the skimage-based lesson by Data Carpentry.

**Comment by mkcor** [comment]
@GParolini would you like to take up this issue? :slightly_smiling_face:

**Comment by mkcor** [comment]
@GParolini that's wonderful! Thank you so much.

**Comment by stefanv** [comment]
@mkcor How much progress has been made on this?

**Comment by mkcor** [comment]
@stefanv non... @GParolini would you like / would you be available to resume working on this?

**Comment by GParolini** [comment]
@mkcor I can take it up again

**Comment by mkcor** [comment]
Wonderful, @GParolini! Keep us posted.

**Comment by decorouz** [comment]
> @mkcor I can take it up again

Hi @GParolini I would like to contribute to this. What examples are currently working on? I suggest a splitting of the gallery examples

**Comment by lagru** [comment]
I'd suggest to just pick one and announce here which one you picked. I hope it's pretty unlikely that there will be overlap if it's communicated in advance. :crossed_fingers:

**Comment by GParolini** [comment]
> > @mkcor I can take it up again
> 
> Hi @GParolini I would like to contribute to this. What examples are currently working on? I suggest a splitting of the gallery examples

Hi @decorouz, thank you for volunteering to work on this issue. The structure of the doc folder is doc/examples/some_folder_name. In March all the code in the folder **applications** was updated. I am planning now to work on the folder **segmentation**. There are nine other folders left to check. Feel free to pick those you prefer. As suggested by @lagru, just post a comment on the one(s) you are working on.

**Comment by somnus-J-307** [comment]
Hi, I'd like to update doc/examples/transform/plot_radon_transform.py for #7330. I plan to replace the remaining stateful pyplot calls in the filter-response plot with Matplotlib's object-oriented interface, without changing the example output.

**Comment by mkcor** [comment]
@somnus-J-307 please go ahead! Thank you very much. 🙏
